src/provider/provider_os_memory.c

```diff
@@ -794,6 +837,7 @@ static umf_result_t os_allocation_merge(void *provider, void *lowPtr,
 
 typedef struct os_ipc_data_t {
     int pid;
+    char shm_name[MAX_NAME];
```

Comment:
You are extending `os_ipc_data_t` by `MAX_NAME` bytes, but it is needed only if named shared memory is used. Perhaps we need two IPC data structures for anonymous and named shared memory.

Reply:
Done

Reply:
Please look into this: https://www.geeksforgeeks.org/flexible-array-members-structure-c/

the common approach is not to specify the array size. And the `os_get_ipc_handle_size` function should be changed accordingly:
```
if (os_provider->shm_name[0]) {
        // os_ipc_data_t->shm_name will be used
        *size = sizeof(os_ipc_data_t) + MAX_NAME;
    } else {
        // os_ipc_data_t->shm_name will NOT be used
        *size = sizeof(os_ipc_data_t);
    }
```

Reply:
@vinser52 I know that. This approach is used when a length of the array is variable. However it is constant in our case, so IMHO the current approach is better in our case.